Felix flo draft

mdhanna opened the pull request.

FlorianNachtigall committed:
Add h3 grid, xgboost, explainability, and some tasks

FlorianNachtigall committed:
Improve description of grid cell aggregation

FlorianNachtigall committed:
Add task solutions and improve task descriptions

FlorianNachtigall committed:
Add section on how to persist model weights

wagnerfe committed:
Adjusted data analysis part and some formal requirements.
Analysis of data in different time windows is skipped for now
as it might distrat from the ML focus. If needed, we can add
a visual analysis (similar to previous year) of spatial trip
distributions for different time windows as another
(easy) exercise.

FlorianNachtigall committed:
Finalize task solutions, organize imports, adapt file paths to support colab

FlorianNachtigall committed:
Colab test run

FlorianNachtigall committed:
Remove leftover mentions of the kernel density approach

FlorianNachtigall committed:
Add follow-up analysis to identifying new bike rack locations

FlorianNachtigall committed:
Add r2 summary table for model comparison

mdhanna commented:
@FlorianNachtigall @wagnerfe Thanks for updating this tutorial! 

Ideally the notebook should run end-to-end even if the exercises are not complete. I see there is a dependency under the Results section where `y_pred_xgb` was defined in Exercise 1. Would you mind commenting out lines 10-12 in that cell, maybe with an instruction to uncomment them if Exercise 1 is complete? Ditto for the next cell in that section.

FlorianNachtigall commented:
Hi @mdhanna,

If you want the whole notebook to run end-to-end, should we comment out all the exercise cells with `## YOUR CODE HERE` since they won't run without changes?

Regarding `y_pred_xgb`: It is also defined in the solution to Exercise 1, so as long as this cell is run, the plotting cell in the results section you mentioned should run just fine.

mdhanna commented:
You don't need to comment out the exercise cells. It's more about being able to execute the entire notebook even if they're unwilling to complete the exercises. I wouldn't expect students to still run the solution cell if they decide they don't want to engage with the exercises.

FlorianNachtigall committed:
Comment out usages of y_pred_xgb

In order to allow to run the notebook without working on any exercises,
comment out the y_pred_xgb variable, which was defined in exercise 1.

cc @mdhanna
